packages/portal/spec-renderer/src/components/SpecDetails.cy.ts

```diff
@@ -0,0 +1,88 @@
+// Cypress component test spec file
+
+import { OperationListItem } from '../types'
+import { jsonSpec, specOpItem } from '../../fixtures/spec-data'
+import SpecDetails from './SpecDetails.vue'
+
+describe('<SpecDetails />', () => {
+  it('renders correctly with props', () => {
+    cy.mount(SpecDetails, {
+      props: {
+        document: jsonSpec as any,
+      },
+    })
+
+    cy.getTestId('kong-public-ui-spec-details-swagger').should('be.visible')
+    cy.getTestId('default-styles').should('exist')
+  })
+
+  it('renders custom styles for swagger-ui for props: hasSidebar', () => {
+    cy.mount(SpecDetails, {
+      props: {
+        document: jsonSpec as any,
+        hasSidebar: true,
+      },
+    })
+
+    cy.getTestId('kong-public-ui-spec-details-swagger').should('be.visible')
+    cy.getTestId('default-styles').should('exist')
+    cy.get('.swagger-ui.has-sidebar').should('be.visible')
+  })
+
+  it('renders custom styles for swagger-ui for props: essentialsOnly', () => {
+    cy.mount(SpecDetails, {
+      props: {
+        document: jsonSpec as any,
+        essentialsOnly: true,
+      },
+    })
+
+    cy.getTestId('kong-public-ui-spec-details-swagger').should('be.visible')
+    cy.getTestId('default-styles').should('exist')
+    cy.getTestId('hide-essentials-styles').should('exist')
+  })
+
+  it('renders custom styles for swagger-ui for props: relativeSidebar', () => {
+    cy.mount(SpecDetails, {
+      props: {
+        document: jsonSpec as any,
+        hasSidebar: true,
+        essentialsOnly: true,
+        relativeSidebar: true,
+      },
+    })
+
+    cy.getTestId('kong-public-ui-spec-details-swagger').should('be.visible')
+    cy.getTestId('default-styles').should('exist')
+    cy.get('.swagger-ui.has-sidebar').should('be.visible')
+    cy.getTestId('hide-essentials-styles').should('exist')
+    cy.getTestId('relative-sidebar-styles').should('exist')
+  })
+
+  it('emits "active-operation" event', () => {
+    const activeOp:OperationListItem = specOpItem
+
+    cy.mount(SpecDetails, {
+      props: {
+        document: jsonSpec as any,
+        activeOperation: activeOp,
+      },
+    })
+
+    cy.getTestId('kong-public-ui-spec-details-swagger').should('be.visible')
+    cy.getTestId('default-styles').should('exist')
+
+    cy.getTestId('kong-public-ui-spec-details-swagger').then(() => {
+      // Check for emitted event
+      // @ts-ignore - vueWrapper DOES exist
+      cy.wrap(Cypress.vueWrapper.emitted()).should('have.property', 'active-operation')
```

Comment:
@evanlindsey heres an example of what I was talking about

Reply:
@adamdehaven we need to add `"cypress/vue"` to this types array: https://github.com/Kong/public-ui-components/blob/main/packages/portal/spec-renderer/tsconfig.json#L7

It works in `khcp-ui` because of this import: https://github.com/Kong/khcp-ui/blob/main/client/cypress/support/commands.ts#L26

If you `import 'cypress/vue'` or any of its exports in a file, it will recognize the types.

Reply:
Ah sweet let me try adding that in one of the global tsconfig files instead of here locally

Reply:
@kaiarrowood I added this on the `main` branch - if you rebase from main, you shouldn't need the `ts-ignore` for `Cypress.vueWrapper`